metaseq/distributed/fully_sharded_data_parallel.py

```diff
@@ -104,6 +104,7 @@ def fsdp_enable_wrap(
         compute_dtype = torch.float32
     fsdp_config = {
         "process_group": group,
+        "process_group_reduce_scatter": group,
```

Comment:
Rather than depend on the default value defined by fairscale, here we are ensuring process_group_reduce_scatter group is the same as the overall process group.

Reply:
Depending on the default value sometimes leads to unpredictable behaviour, where the world size of the process group and the reduce scatter process group differ.

Reply:
https://github.com/facebookresearch/fairscale/blob/main/fairscale/nn/data_parallel/fully_sharded_data_parallel.py#L361

Reply:
Hm, our branch of fairscale supposedly fixed this already... But maybe it's better to handle it here for us.